src/Skia/Avalonia.Skia/Gpu/Metal/SkiaMetalApi.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Avalonia.Platform.Interop;
+using SkiaSharp;
+using BindingFlags = System.Reflection.BindingFlags;
+
+namespace Avalonia.Skia.Metal;
+
+// We use reflection only on member that are guaranteed to be referenced by OpenGL gpu code, so there should not
+// be problems with trimming
+
+internal unsafe class SkiaMetalApi
+{
+    //    internal static extern unsafe IntPtr gr_direct_context_make_metal_with_options(
+    //void* device,
+    //void* queue,
+    //    GRContextOptionsNative* options);
+    // device, queue, options
+    delegate* unmanaged[Stdcall] <IntPtr, IntPtr, IntPtr, IntPtr> _gr_direct_context_make_metal_with_options;
+    // width, height, samples, &info
+    private delegate* unmanaged[Stdcall]<int, int, int, GRMtlTextureInfoNative*, IntPtr>
+        _gr_backendrendertarget_new_metal;
+    private readonly ConstructorInfo _contextCtor;
+    private readonly MethodInfo _contextOptionsToNative;
+    private readonly ConstructorInfo _renderTargetCtor;
+
+
+    public SkiaMetalApi()
+    {
+        // Make sure that skia is loaded
+        GC.KeepAlive(new SKPaint());
+        
+        var loader = AvaloniaLocator.Current.GetRequiredService<IDynamicLibraryLoader>();
+#if NET6_0_OR_GREATER
+        var dll = NativeLibrary.Load("libSkiaSharp", typeof(SKPaint).Assembly, null);
+#else
+        var dll = loader.LoadLibrary("libSkiaSharp");
+#endif
+        _gr_direct_context_make_metal_with_options = (delegate* unmanaged[Stdcall] <IntPtr, IntPtr, IntPtr, IntPtr>)
+            loader.GetProcAddress(dll, "gr_direct_context_make_metal_with_options", false);
+        _gr_backendrendertarget_new_metal =
+            (delegate* unmanaged[Stdcall]<int, int, int, GRMtlTextureInfoNative*, IntPtr>)
+            loader.GetProcAddress(dll, "gr_backendrendertarget_new_metal", false);
+        
+        _contextCtor = typeof(GRContext).GetConstructor(
+            BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null,
+            new[] { typeof(IntPtr), typeof(bool) }, null) ?? throw new MissingMemberException("GRContext.ctor(IntPtr,bool)");
+
+                
+        _renderTargetCtor = typeof(GRBackendRenderTarget).GetConstructor(
+            BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null,
+            new[] { typeof(IntPtr), typeof(bool) }, null) ?? throw new MissingMemberException("GRContext.ctor(IntPtr,bool)");
+
+        _contextOptionsToNative = typeof(GRContextOptions).GetMethod("ToNative",
+                                      BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
+                                  ?? throw new MissingMemberException("GRContextOptions.ToNative()");
```

Comment:
To avoid trimming issues, please add these attributes on SkiaMetalApi ctor.
```
[DynamicDependency(DynamicallyAccessedMemberTypes.NonPublicConstructors, typeof(GRContext))]
[DynamicDependency(DynamicallyAccessedMemberTypes.NonPublicConstructors, typeof(GRBackendRenderTarget))]
[DynamicDependency(DynamicallyAccessedMemberTypes.NonPublicMethods, typeof(GRContextOptions))]
```

Reply:
Have not tested it, but should work without any rd.xml iles.

Reply:
It's also possible to write a specific member signature in the DynamicDependency attribute, but only if you know exactly one needed.

Reply:
That's just to ensure that "We use reflection only on member that are guaranteed to be referenced by OpenGL gpu code" is actually not a problem, as I have doubts. Something like GRContextOptions.ToNative could be trimmed out otherwise.